p2p/server.py

```diff
@@ -0,0 +1,125 @@
+import asyncio
+import os
+import logging
+import sys
+
+from typing import Tuple
+from eth_keys import (
+    datatypes,
+    keys,
+)
+from eth_utils import (
+    decode_hex,
+    keccak,
+)
+
+from p2p.auth import (
+    HandshakeResponder,
+    decode_auth_plain,
+    decode_auth_eip8,
+)
+from p2p.cancel_token import CancelToken
+from p2p.constants import (
+    ENCRYPTED_AUTH_MSG_LEN,
+    HASH_LEN,
+)
+from p2p.ecies import ecdh_agree
+from p2p.kademlia import (
+    Address,
+    Node,
+)
+from p2p.utils import sxor
+
+
+class Server:
+    """Server listening for incoming connections"""
+    logger = logging.getLogger("p2p.server.Server")
+    _server_address = ('localhost', 10000)
+    _connect_loop_sleep = 2
+    
+    def __init__(self, privkey: datatypes.PrivateKey) -> None:
+        self.privkey = privkey
+        self.cancel_token = CancelToken('Server')
+        self.incoming_connections = []
```

Comment:
I think that instead of this `list` based storage we'll want the `Server` to be given a `PeerPool` that it can push new peers into.

Reply:
@pipermerriam The list is a temp hack - It looks like implementing `registerPeer` is not insignificant i thought it would be better off in a following pr - to avoid this pr growing too large? I've added a peer_pool to the `Server` object. If you think the `registerPeer` logic would be better implemented in this pr - i can go ahead and do that